packages/xstate-graph/src/graph.ts

```diff
@@ -67,59 +73,49 @@ export function getChildren(stateNode: AnyStateNode): AnyStateNode[] {
   return children;
 }
 
-export function serializeState<TContext>(state: State<TContext, any>): string {
-  const { value, context } = state;
-  return context === undefined
-    ? JSON.stringify(value)
-    : JSON.stringify(value) + ' | ' + JSON.stringify(context);
+export function serializeMachineState(state: AnyState): SerializedState {
+  const { value, context, actions } = state;
+  return JSON.stringify({ value, context, actions }) as SerializedState;
 }
 
 export function serializeEvent<TEvent extends EventObject>(
   event: TEvent
-): string {
-  return JSON.stringify(event);
-}
-
-export function deserializeEventString<TEvent extends EventObject>(
-  eventString: string
-): TEvent {
-  return JSON.parse(eventString) as TEvent;
+): SerializedEvent {
+  return JSON.stringify(event) as SerializedEvent;
 }
 
 const defaultValueAdjMapOptions: Required<ValueAdjMapOptions<any, any>> = {
   events: {},
   filter: () => true,
-  stateSerializer: serializeState,
+  stateSerializer: serializeMachineState,
   eventSerializer: serializeEvent
 };
 
-function getValueAdjMapOptions<TContext, TEvent extends EventObject>(
-  options?: ValueAdjMapOptions<TContext, TEvent>
-): Required<ValueAdjMapOptions<TContext, TEvent>> {
+function getValueAdjMapOptions<TState, TEvent extends EventObject>(
```

Comment:
Shouldn't this have a constraint like this?
```suggestion
function getValueAdjMapOptions<TState extends AnyState, TEvent extends EventObject>(
```